docs/mllib-migration-guides.md

```diff
@@ -7,6 +7,25 @@ description: MLlib migration guides from before Spark SPARK_VERSION_SHORT
 
 The migration guide for the current Spark version is kept on the [MLlib Programming Guide main page](mllib-guide.html#migration-guide).
 
+## From 1.3 to 1.4
```

Comment:
No content change here. Just moved the paragraphs from `mllib-guide` and `ml-guide`.